src/hotspot/share/gc/shenandoah/shenandoahHeap.hpp

```diff
@@ -354,6 +354,9 @@ class ShenandoahHeap : public CollectedHeap {
   // effort.  If this happens, the requesting thread blocks until some other thread manages to evacuate the offending object.
   // Only after "all" threads fail to evacuate an object do we consider the evacuation effort to have failed.
 
+  // How many full-gc cycles have been completed?
+  volatile size_t _completed_fullgc_cycles;
```

Comment:
`ShenandoahCollectorPolicy` already counts full GC cycles (implicit and  explicit). We could add a `get_full_gc_count` there without duplicating any functionality.

Reply:
good idea.  I searched around and couldn't figure out where this was.  I'll pursue this simplification.

Reply:
I've made this change as suggested by you.